go/binlog/gomysql_reader.go

```diff
@@ -160,10 +158,6 @@ func (this *GoMySQLReader) StreamEvents(canStopStreaming func() bool, entriesCha
 }
 
 func (this *GoMySQLReader) Close() error {
-	// Historically there was a:
-	//   this.binlogSyncer.Close()
-	// here. A new go-mysql version closes the binlog syncer connection independently.
-	// I will go against the sacred rules of comments and just leave this here.
-	// This is the year 2017. Let's see what year these comments get deleted.
```

Comment:
@shlomi-noach I noticed from my testing that this was required to stop the binlog syncer from running indefinitely. i'm not sure if this comment is specific to running this code as a process that exits, but it seems this is necessary from my end. would appreciate any context you have on this, and if this change looks right to you

Reply:
I do get the following error in the log:
```
2017/12/18 11:51:15 binlogsyncer.go:126: [info] syncer is closing...
2017/12/18 11:51:15 binlogsyncer.go:554: [error] connection was bad
2017/12/18 11:51:15 binlogstreamer.go:47: [error] close sync with err: Sync was closed
2017/12/18 11:51:15 binlogsyncer.go:141: [info] syncer is closed
```

But it _seems_ to not cause any issues.

If I don't have this explicit close, the binlogsyncer lingers on after the migration has finished:
```
2017-12-18 11:53:40 INFO Done migrating `zenpayroll_development`.`companies`
2017-12-18 11:53:40 INFO Removing socket file: /gh-ost/gh-ost.test.sock
2017-12-18 11:53:40 INFO Tearing down inspector
2017-12-18 11:53:40 INFO Tearing down applier
2017-12-18 11:53:40 INFO Tearing down streamer
2017/12/18 11:55:00 binlogsyncer.go:632: [info] rotate to (mysql-bin-changelog.001359, 4)
2017/12/18 11:55:00 binlogsyncer.go:632: [info] rotate to (mysql-bin-changelog.001359, 4)
2017/12/18 11:55:00 binlogsyncer.go:632: [info] rotate to (mysql-bin-changelog.001359, 4)
2017/12/18 12:00:00 binlogsyncer.go:632: [info] rotate to (mysql-bin-changelog.001360, 4)
2017/12/18 12:00:00 binlogsyncer.go:632: [info] rotate to (mysql-bin-changelog.001360, 4)
```